lib/upgrade/etcd/conversionv1v3/profile_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conversionv1v3
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	"github.com/projectcalico/libcalico-go/lib/apis/v1/unversioned"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+)
+
+var profileTable = []struct {
+	description string
+	v1API       unversioned.Resource
+	v1KVP       *model.KVPair
+	v3API       apiv3.Profile
+}{
+	{
+		description: "fully populated Profile",
+		v1API: apiv1.Profile{
+			Metadata: apiv1.ProfileMetadata{
+				Name:   "nameyMcProfileName",
+				Tags:   []string{"meep", "mop"},
+				Labels: map[string]string{"thingy": "val"},
+			},
+			Spec: apiv1.ProfileSpec{
+				IngressRules: []apiv1.Rule{V1InRule1, V1InRule2},
+				EgressRules:  []apiv1.Rule{V1EgressRule1, V1EgressRule2},
+			},
+		},
+		v1KVP: &model.KVPair{
+			Key: model.ProfileKey{
+				Name: "nameyMcProfileName",
+			},
+			Value: &model.Profile{
+				Rules: model.ProfileRules{
+					InboundRules:  []model.Rule{V1ModelInRule1, V1ModelInRule2},
+					OutboundRules: []model.Rule{V1ModelEgressRule1, V1ModelEgressRule2},
+				},
+				Tags:   []string{"meep", "mop"},
+				Labels: map[string]string{"thingy": "val"},
+			},
+		},
+		v3API: apiv3.Profile{
+			ObjectMeta: v1.ObjectMeta{
+				Name: "nameymcprofilename",
+			},
+			Spec: apiv3.ProfileSpec{
+				Ingress:       []apiv3.Rule{V3InRule1, V3InRule2},
+				Egress:        []apiv3.Rule{V3EgressRule1, V3EgressRule2},
+				LabelsToApply: map[string]string{"thingy": "val", "meep": "", "mop": ""},
```

Comment:
Could you add the prefix `pncs.` to one of these label keys? Just for more realistic labels.